v2/assistantv1/assistant_v1.go

```diff
@@ -253,11 +256,13 @@ func (assistant *AssistantV1) MessageWithContext(ctx context.Context, messageOpt
 	if err != nil {
 		return
 	}
-	err = core.UnmarshalModel(rawResponse, "", &result, UnmarshalMessageResponse)
```

Comment:
What is the reason fo these strange changes?

Reply:
I wouldn't call the changes strange - it adds another level of `nil` checking when unmarhsalling to ensure that we only attempt an unmarshal if `rawResponse` is a non-nil value.